### ISSUE TITLE
3.8.0 uninstallable

### ISSUE THREAD

**Issue by jpds** [mirage/mirage#1182]
I'm trying to install mirage on a fresh Debian 10 VM, but it seems that while 3.8.0 is available at https://opam.ocaml.org/packages/mirage/ - `opam` can't seem to install it:

```
$ opam install 'mirage>=3.8.0' 
Sorry, no solution found: there seems to be a problem with your request.

No solution found, exiting
$ opam install 'mirage>=3.7.0' 
Sorry, no solution found: there seems to be a problem with your request.

No solution found, exiting
$ opam install 'mirage>=3.6.0' 
The following actions will be performed:
  ∗ install cmdliner                 1.0.4    [required by functoria]
  ∗ install conf-m4                  1        [required by ocamlfind]
  ∗ install ocaml-secondary-compiler 4.08.1-1 [required by ocamlfind-secondary]
  ∗ install ocamlbuild               0.14.0   [required by astring, bos]
  ∗ install ocamlfind                1.8.1    [required by astring, bos]
  ∗ install topkg                    1.0.1    [required by astring, bos]
  ∗ install ocamlgraph               1.8.8    [required by functoria]
  ∗ install ocamlfind-secondary      1.8.1    [required by dune]
  ∗ install astring                  0.8.4    [required by mirage]
  ∗ install dune                     2.6.1    [required by mirage]
  ∗ install stdlib-shims             0.1.0    [required by fmt]
  ∗ install sexplib0                 v0.14.0  [required by ipaddr]
  ∗ install seq                      0.2.2    [required by fmt]
  ∗ install result                   1.5      [required by rresult, fpath,
                                              ptime]
  ∗ install macaddr                  4.0.0    [required by ipaddr]
  ∗ install fmt                      0.8.8    [required by functoria, bos,
                                              mirage-runtime]
  ∗ install rresult                  0.6.0    [required by bos, functoria]
  ∗ install ptime                    0.8.5    [required by functoria]
  ∗ install fpath                    0.7.2    [required by bos, functoria]
  ∗ install logs                     0.7.0    [required by mirage]
  ∗ install functoria-runtime        3.0.3    [required by mirage-runtime]
  ∗ install domain-name              0.3.0    [required by ipaddr]
  ∗ install bos                      0.2.0    [required by mirage]
  ∗ install ipaddr                   4.0.0    [required by mirage]
  ∗ install functoria                3.1.1    [required by mirage]
  ∗ install mirage-runtime           3.6.0    [required by mirage]
  ∗ install mirage                   3.6.0
===== ∗ 27 =====
Do you want to continue? [Y/n]
```

**Comment by jpds** [comment]
I just spotted that mirage 3.8 depends on ocaml 4.06 and Debian has 4.05.

Ran this, and then it seems to work for me:

```
$ opam switch create 4.10.0
$ eval $(opam env)
$ opam install 'mirage>=3.8.0'
```

**Comment by dinosaure** [comment]
Can you try `opam install mirage.3.8.0`. I'm not sure that the syntax is "valid". On my side, I can not reproduce your state:
```shell
dinosaure@turbine:~$ opam switch create test 4.10.0

<><> Gathering sources ><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
[ocaml-base-compiler.4.10.0] found in cache

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
∗ installed base-bigarray.base
∗ installed base-threads.base
∗ installed base-unix.base
∗ installed ocaml-base-compiler.4.10.0
∗ installed ocaml-config.1
∗ installed ocaml.4.10.0
Done.
# Run eval $(opam env) to update the current shell environment
dinosaure@turbine:~$ eval $(opam env)
dinosaure@turbine:~$ opam install mirage.3.8.0
The following actions will be performed:
  ∗ install dune              2.6.1  [required by mirage]
  ∗ install seq               base   [required by fmt]
  ∗ install cmdliner          1.0.4  [required by functoria]
  ∗ install conf-m4           1      [required by ocamlfind]
  ∗ install ocamlbuild        0.14.0 [required by astring, bos]
  ∗ install stdlib-shims      0.1.0  [required by mirage]
  ∗ install result            1.5    [required by rresult, fpath, ptime]
  ∗ install mmap              1.1.0  [required by lwt]
  ∗ install macaddr           5.0.0  [required by ipaddr]
  ∗ install dune-private-libs 2.6.1  [required by dune-configurator]
  ∗ install cppo              1.6.6  [required by lwt]
  ∗ install ocamlfind         1.8.1  [required by astring, bos]
  ∗ install dune-configurator 2.6.1  [required by lwt]
  ∗ install topkg             1.0.1  [required by astring, bos]
  ∗ install ocamlgraph        1.8.8  [required by functoria]
  ∗ install base-bytes        base   [required by ocplib-endian]
  ∗ install rresult           0.6.0  [required by bos, functoria]
  ∗ install ptime             0.8.5  [required by functoria]
  ∗ install fmt               0.8.8  [required by functoria, bos, mirage-runtime]
  ∗ install astring           0.8.4  [required by mirage]
  ∗ install ocplib-endian     1.1    [required by lwt]
  ∗ install functoria-runtime 3.0.3  [required by mirage-runtime]
  ∗ install fpath             0.7.2  [required by bos, functoria]
  ∗ install domain-name       0.3.0  [required by ipaddr]
  ∗ install lwt               5.3.0  [required by mirage-runtime]
  ∗ install ipaddr            5.0.0  [required by mirage]
  ∗ install logs              0.7.0  [required by mirage]
  ∗ install mirage-runtime    3.8.0  [required by mirage]
  ∗ install bos               0.2.0  [required by mirage]
  ∗ install functoria         3.1.1  [required by mirage]
  ∗ install mirage            3.8.0
===== ∗ 31 =====
Do you want to continue? [Y/n] 
```

**Comment by dinosaure** [comment]
Closed, `mirage` requires at least 4.06 indeed.
